crates/pathfinder/Cargo.toml

```diff
@@ -20,7 +20,7 @@ anyhow = "1.0.44"
 async-trait = "0.1.52"
 base64 = "0.13.0"
 # paritys scale codec locks us here
-bitvec = "0.20.4"
+bitvec = { version = "0.20.4", features = ["serde"] }
```

Comment:
Something to consider here - what should these edge node paths be defined as in the RPC output? What formatting should they ideally have?

The format should be easily parseable by most starknet/web3 related SDKs.

Some options:

1. A `StarkHash` i.e. a Felt. The bonus here is that most things in the RPC are this already. Will need to the bit length encoded separately though.
2. Whatever `bitvec` does. A negative here is that we aren't in control of how this is serialized. So if `bitvec` ever changes this we'll have to make a plan.
3. Something else - varint maybe?

I personally would recommend (1) just because its what the ecosystem is used to. I'm also unsure what `bitvec` actually does encoding-wise?

Reply:
So `bitvec` calls the [`as_bitslice()`](https://docs.rs/bitvec/latest/src/bitvec/serdes/slice.rs.html#99) function and then serializes each fields as so: [code](https://docs.rs/bitvec/0.22.3/src/bitvec/serdes.rs.html#96-103) (basically, an array of bytes).

An example of the actual output (courtesy of @FawadHa1der :D): http://sprunge.us/M7EOWR

I do agree it would be easier to have `path` represented as a `StarkHash`. Should we serialize the `EdgeProof` struct like so:
```rust
{ "Edge" : { "path": { "value": "0x123456...", "len": "42" }, "child_hash": "0x4242"}}
```
? The main difference being: removing `head`, renaming `data` to `value`, renaming `bits` to `len`, and representing `value` as a `Starkhash` instead of an array of bytes.

Reply:
Yeah that looks good to me 👍

Reply:
72d53c759971286c029d859579df95cb24181833 and 51bc0aadf2c4c4326370f65f2054b95022664ef9 :)

Ended up impl serialize in the `get_proof.rs` file. It adds some two wrapper structs (I couldn't find a better way of doing this with serde... went to look for help in the Rust Community Discord [link](https://discord.com/channels/273534239310479360/1048181382184173609/1048181382184173609) but no better solution was found :p).